tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -39,13 +39,28 @@ function get_container_runtime() {
                 die "invalid node name"
 	fi
 	if echo "$runtime" | grep -qE 'containerd.*-k3s'; then
+		echo $(k3s_service_name)
+	else
+		echo "$runtime" | awk -F '[:]' '{print $1}'
+	fi
+}
+
+function k3s_service_name() {
+	local k3s_service=""
+	if [ -x /sbin/openrc-run ]; then
+		k3s_service=$(ls /etc/init.d/ | grep k3s)
```

Comment:
That's an awkward one. I think the best you could do is make this slightly more reliable by anchoring "k3b". Something like this:

```suggestion
		k3s_service=$(ls /etc/init.d/ | grep -E "\<k3s\>")
```

... which will still match for a service name like `k3s-service`.

Reply:
Totally agree, a better solution need a big re-fact, like chroot based solution mentioned above, I guess.